core/vm/interface.go

```diff
@@ -42,7 +42,7 @@ type StateDB interface {
 	SetCode(common.Address, []byte)
 	GetCodeSize(common.Address) int
 
-	ValidatorWrapper(common.Address) (*staking.ValidatorWrapper, error)
+	ValidatorWrapper(common.Address, bool) (*staking.ValidatorWrapper, error)
```

Comment:
We should take the three validator methods , assuming you do the two named book methods , and factor them out into separate interface , then embed into this one , then we can more easily test just staking validator related state changes

Reply:
clarify your suggestion? I don't get it.

Reply:
like:

```go
type ValidatorUpdater interface {
  WrapperBoolTrue(common.Address)
  WrapperBoolFalse(common.Addres)
  UpsertValidator(common.Addrsss...)
} 


type StateDB struct {
  ValidatorUpdater
} 
```

Something like that, but not a big deal if not done

Reply:
where the booltrue/boolfalse are better named to reflect what the bool param does.

Reply:
ok, agree.